main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -601,7 +600,7 @@ private IConnector getConnector() {
     }
 
     public boolean supportsRefresh() {
-        return getConnector() instanceof ISearchByGeocode && getConnector() instanceof ILogin;
+        return getConnector() instanceof ISearchByGeocode;
```

Comment:
Oh, and `ISearchByGeocode` is also implemented for `InternalConnector` so we should probably add a hard-coded exception for it.

Reply:
` return getConnector() instanceof ISearchByGeocode && !InternalConnector.getInstance().equals(getConnector());` should be suitable (tested just now)

Reply:
Ok, the change is trivial, but what does it do? What state is expressed by the right part of the `&&`? Perhaps the answer lies in knowing what the purpose of the InternalConnector is?

Reply:
`Internal Connector` is used for creating "user defined caches" inside c:geo via long-tab on map (a bit like waypoints, which belong to no cache and can be used for tagging nearly everything). Since local caches can't be refreshed, they should be excluded...

![grafik](https://user-images.githubusercontent.com/64581222/117039823-830e5e80-ad09-11eb-9ef0-aa5ccc02c321.png)

Reply:
Ah ok, and they support a `ISearchByGeocode` but can't be refreshed which is why the refresh function must not be bound to the UI, right? So just leaving out the condition on the rights side of the `&&` would have broken it for UDC aka ZZ-caches.

Reply:
Yes, nothing really bad would happen though, but we would have a useless button for UDC-caches (as `Internal Connector` was the only one, which is instance of `ISearchByGeocode `but not `ILogin`)

Reply:
Related to this subject is a discussion I had started in the team channel, it is mainly directed at @Lineflyer but it appears to me that he doesn't get notified on new content in those channels. If @Lineflyer is reading this here: Pls go check the internal channel for a discussion on the topic at hand here. It is about showing actual "found state", a feature which will be high on demand as soon as the first ALC functionality in cgeo hits the road.

Reply:
My PC was broken, now the new one arrived and I am "alive" again. :)
Will check back the team chat as time permits.

Reply:
Still cannot understand why we do try to infer the "connector is able to refresh cache date" from its "supports search by geocode" and "can login" capabilities. That may have made sense at the time where the connector framework has been implemented, but as we see now, that's no longer valid. So I would vote for a (IMHO) more clean solution by explicitly adding the capability

by either adding something like a "IRefreshable" capability to all connectors supporting a refresh.

or by adding a method like `canRefresh()` to the `AbstractConnector`, defaulting to `true`, and overwriting this method in all connectors that can't.

`Geocache.canRefresh()` needs to be adapted to either checking the connector for `instanceOf IRefreshable` or for `canRefresh() == true`.

(Whether `canRefresh()` has a fixed value per connector or even checks login state would be up to us.)

Reply:
OTOH "supports search by geocode" means that cache details can be fetched via a geocode, which implies that the cacheobject get updated via internet, which always will be the case for all potentially following connectors except the `InternalConnector`. 

> `canRefresh()` to the `AbstractConnector`, defaulting to `true`

This would be fine as well, but an `IRefreshable` is IMHO definitely to much